src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -138,6 +140,41 @@ double JacobianPerformanceTest(ModuleType& module,
   return arma::max(arma::max(arma::abs(centralDifference - delta)));
 }
 
+// Simple numerical gradient checker.
```

Comment:
Just to make sure I understand, the goal here is to ensure that the gradient that's calculated and returned is close to an approximation of the gradient calculated by perturbing the inputs?